com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/flowable/FlowableFacade.java

```diff
@@ -230,6 +235,13 @@ public void deleteProcessInstance(String processInstanceId, String deleteReason)
                 // different if the process has parallel executions.
                 processEngine.getRuntimeService()
                              .setVariable(processInstanceId, Constants.PROCESS_ABORTED, Boolean.TRUE);
+                while (true) {
+                    List<Execution> allLeafExecutions = getAllLeafExecutions(processInstanceId);
+
+                    if (haveAllLeafsHaveDeadLetterJobs(allLeafExecutions)) {
+                        break;
+                    }
```

Comment:
What if there is a bug in the code? We would stay in the loop forever.
Maybe giving it some time would be enough in order the executions to finish. For instance, 10 mins MAX or more?

Reply:
Thanks for the idea!
Done